diff --git a/libavformat/movenc.c b/libavformat/movenc.c
index ec09e90436..53e68ceefb 100644
--- a/libavformat/movenc.c
+++ b/libavformat/movenc.c
@@ -121,6 +121,7 @@ static const AVOption options[] = {
     { "use_editlist", "use edit list", offsetof(MOVMuxContext, use_editlist), AV_OPT_TYPE_BOOL, {.i64 = -1}, -1, 1, AV_OPT_FLAG_ENCODING_PARAM},
     { "use_stream_ids_as_track_ids", "use stream ids as track ids", offsetof(MOVMuxContext, use_stream_ids_as_track_ids), AV_OPT_TYPE_BOOL, {.i64 = 0}, 0, 1, AV_OPT_FLAG_ENCODING_PARAM},
     { "video_track_timescale", "set timescale of all video tracks", offsetof(MOVMuxContext, video_track_timescale), AV_OPT_TYPE_INT, {.i64 = 0}, 0, INT_MAX, AV_OPT_FLAG_ENCODING_PARAM},
+    { "audio_track_timescale", "set timescale of all audio tracks", offsetof(MOVMuxContext, audio_track_timescale), AV_OPT_TYPE_INT, {.i64 = 0}, 0, INT_MAX, AV_OPT_FLAG_ENCODING_PARAM },
     { "write_btrt", "force or disable writing btrt", offsetof(MOVMuxContext, write_btrt), AV_OPT_TYPE_BOOL, {.i64 = -1}, -1, 1, AV_OPT_FLAG_ENCODING_PARAM},
     { "write_prft", "Write producer reference time box with specified time source", offsetof(MOVMuxContext, write_prft), AV_OPT_TYPE_INT, {.i64 = MOV_PRFT_NONE}, 0, MOV_PRFT_NB-1, AV_OPT_FLAG_ENCODING_PARAM, .unit = "prft"},
       { "pts", NULL, 0, AV_OPT_TYPE_CONST, {.i64 = MOV_PRFT_SRC_PTS}, 0, 0, AV_OPT_FLAG_ENCODING_PARAM, .unit = "prft"},
@@ -6831,7 +6832,13 @@ static int mov_init(AVFormatContext *s)
                     return AVERROR(ENOMEM);
             }
         } else if (st->codecpar->codec_type == AVMEDIA_TYPE_AUDIO) {
-            track->timescale = st->codecpar->sample_rate;
+            if (mov->audio_track_timescale) {
+                track->timescale = mov->audio_track_timescale;
+                if (mov->mode == MODE_ISM && mov->audio_track_timescale != 10000000)
+                    av_log(s, AV_LOG_WARNING, "Warning: some tools, like mp4split, assume a timescale of 10000000 for ISMV.\n");
+            } else {
+                track->timescale = st->codecpar->sample_rate;
+            }
             if (!st->codecpar->frame_size && !av_get_bits_per_sample(st->codecpar->codec_id)) {
                 av_log(s, AV_LOG_WARNING, "track %d: codec frame size is not set\n", i);
                 track->audio_vbr = 1;
@@ -6922,8 +6929,10 @@ static int mov_init(AVFormatContext *s)
            doesn't mandate a track timescale of 10,000,000. The muxer allows a custom timescale
            for video tracks, so if user-set, it isn't overwritten */
         if (mov->mode == MODE_ISM &&
-            (st->codecpar->codec_type != AVMEDIA_TYPE_VIDEO ||
-            (st->codecpar->codec_type == AVMEDIA_TYPE_VIDEO && !mov->video_track_timescale))) {
+            ((st->codecpar->codec_type != AVMEDIA_TYPE_AUDIO &&
+                          st->codecpar->codec_type != AVMEDIA_TYPE_VIDEO) ||
+                          (st->codecpar->codec_type == AVMEDIA_TYPE_AUDIO && !mov->audio_track_timescale) ||
+                          (st->codecpar->codec_type == AVMEDIA_TYPE_VIDEO && !mov->video_track_timescale))) {
              track->timescale = 10000000;
         }
 
diff --git a/libavformat/movenc.h b/libavformat/movenc.h
index c9c4b6cf63..da83fcfd60 100644
--- a/libavformat/movenc.h
+++ b/libavformat/movenc.h
@@ -211,6 +211,7 @@ typedef struct MOVMuxContext {
     int first_trun;
 
     int video_track_timescale;
+    int audio_track_timescale;
 
     int reserved_moov_size; ///< 0 for disabled, -1 for automatic, size otherwise
     int64_t reserved_header_pos;
